internal/repository/repository.go

```diff
@@ -822,6 +1001,16 @@ func StreamPack(ctx context.Context, beLoad BackendLoadFn, key *crypto.Key, pack
 			// decryption errors are likely permanent, give the caller a chance to skip them
 			nonce, ciphertext := buf[:key.NonceSize()], buf[key.NonceSize():]
 			plaintext, err := key.Open(ciphertext[:0], nonce, ciphertext, nil)
+			if err == nil && entry.IsCompressed() {
+				if cap(decode) < int(entry.DataLength()) {
+					decode = make([]byte, 0, entry.DataLength())
+				}
```

Comment:
```suggestion
```
I would just leave this out. Since you are using EncodeAll the "frame content size" will be set with the uncompressed data size, so the decoder will do the same as this code. However the soon-to-be-released decoder likes to have a few (for now 16) extra bytes on the output slice, so allocating a precise slice may lead to either a re-alloc or a less optimal code path to be chosen.

You of course couldn't know that, but it would seem beneficial to just leave the potential alloc to the decoder.

Reply:
Good idea, we'll just do that.